Rubberduck.CodeAnalysis/QuickFixes/IntroduceLocalVariableForRedimQuickFix.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using Antlr4.Runtime;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Concrete;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+
+namespace Rubberduck.Inspections.QuickFixes
+{
+    public sealed class IntroduceLocalVariableForRedimQuickFix : QuickFixBase
+    {
+        private readonly RubberduckParserState _state;
+
+        public IntroduceLocalVariableForRedimQuickFix(RubberduckParserState state)
+            : base(typeof(UndeclaredRedimVariableInspection))
+        {
+            _state = state;
+        }
+
+        public override bool CanFixInProcedure => true;
+        public override bool CanFixInModule => true;
+        public override bool CanFixInProject => true;
+
+        public override void Fix(IInspectionResult result)
+        {
+            VBAParser.RedimVariableDeclarationContext redimVariable = FindContext<VBAParser.RedimVariableDeclarationContext>(result.Target.Context);
+            VBAParser.RedimStmtContext redimStatement = FindContext<VBAParser.RedimStmtContext>(redimVariable);
+            var instruction = $"{Environment.NewLine}Dim {result.Target.IdentifierName}() As {result.Target.AsTypeName}{Environment.NewLine}";
```

Comment:
This does not deal with indentation. An Alternative might be:

```
var instruction = redimStatement.GetText().ReplaceFirst("Redim", "Dim");
```

We basically copy the RedimStatement anyways, being explicit about it doesn't hurt :)